app/src/main/java/org/simple/clinic/enterotp/EnterOtpEffectHandler.kt

```diff
@@ -49,6 +49,7 @@ class EnterOtpEffectHandler @AssistedInject constructor(
         .addAction(AllowOtpEntry::class.java, { uiActions.showOtpEntryMode(OtpEntry) }, schedulers.ui())
         .addConsumer(BlockOtpEntryUntil::class.java, { uiActions.showOtpEntryMode(BruteForceOtpEntryLocked(it.blockTill)) }, schedulers.ui())
         .addAction(HideErrors::class.java, { uiActions.hideError() }, schedulers.ui())
+        .addConsumer(ShowIncorrectOtpLimitReachedError::class.java, { uiActions.showLimitReachedError(it.attemptsMade) }, schedulers.ui())
```

Comment:
I feel like we are handling all the UI rendering in UI actions, we should not be doing that. UI actions are mostly for navigation or fire and forget UI operations.